### ISSUE TITLE
Add a shortcut or a button to restart extension host.

### ISSUE THREAD

**Issue by siddhant1** [microsoft/vscode#78166]
Right now if the extension host crashes we have to restart the editor , it would be awesome if we could add a button or shortcut for the same.

**Comment by alexandrudima** [comment]
If the extension host crashes, there is a popup that contains a button to restart it. Have you seen that @siddhant1 ?

**Comment by siddhant1** [comment]
Yup , I have seen that , but sometimes u close that accidentally by removing all notifications and then I notice that my extensions are crashed! The only solution is restart the editor.

**Comment by goldingdamien** [comment]
This crashes for me a lot. I think it would be nice to have information on why it crashes and for a way to restart to be easily done. I'm assuming CPU load is the issue for me as my computer's bottleneck is the CPU.

**Comment by thesayyn** [comment]
It will be more helpful if we put a button to the workspace bar below that will only appear when extension host crashes.

**Comment by kokoshneta** [comment]
> If the extension host crashes, there is a popup that contains a button to restart it. Have you seen that @siddhant1 ?

I’ve never seen that popup, so I’m guessing the extension host has never actually _crashed_ on me.

It very frequently (as in, at least once an hour during normal work) becomes _unresponsive_, making extensions stop working. Or at least that’s what the output tab says (“UNRESPONSIVE extension host, [extension] took XX% of XXXXms, saved PROFILE here…”).

The only way to get the extension back then is to restart the entire editor, as far as I can tell. :-/

**Comment by TryTryAgain** [comment]
I have found the command palette 'Developer: Reload Window' works well for cases where the extension host crashes and you want to reload it without fully exiting and re-opening vscode. See https://stackoverflow.com/a/42002990/805031 for more info.

**Comment by jsuddsjr** [comment]
Even better, could you please restart the extension host automatically? Reporting that it has crashed, and then waiting for me to recognize that my editor is not working because I have to click an OK button is a bad experience. Rather, only notify me if the extension host is crashing repeatedly, and then provide some context as to what was happening when it crashed. Then I could actually do something to fix it (either disable an extension or restart my editor).

**Comment by grzegorzgit** [comment]
It is crazy that it happens, and that it happens so often.

- First, why do you inform me about it in a first place instead of relaunching it immediately after this problem? 
- Secondly, why is this information for only few seconds and then disappears? If I go for a minute away from computer, VS Code does its amazing every-5-minutes-crash, and unless I remember to check, my extensions are not working.
- At third, why there is no information what cause the problem? Exactly which extension is broken? What should I do to avoid in the future? Anything?
- And at last, how to solve it? Restarting VS Code sounds for me the same nice as restarting the computer - so yeah... not so fancy way of **solving** software problem
Sorry Microsoft, but such mistakes  would be explainable for some small IT company who has no idea about UI and UX. Microsoft seems to not be such a company, so what's the problem? You just don't care too much?

**Comment by pcgeek86** [comment]
When the extension host crashes, I typically hit ESCAPE to close any notifications. However, there is no option in the VSCode Command Palette for me to force the extension host to reload. The only option is to use the [slower] `Reload Window` option.

I would also like to see the ability to restart the extension host, from the Command Palette.

**Comment by alexdima** [comment]
🤖 Thanks for the request. This was implemented in https://github.com/microsoft/vscode/commit/d7890fa51fdd3cafa06325c9739608a29cba677c and shipped in VS Code 1.42. Use **Developer: Restart Extension Host** from the Command Palette; you can also assign that command a keyboard shortcut.
